docs/develop/08_opendid/04_integrate_opendid.md

```diff
@@ -0,0 +1,117 @@
+---
+id: integrate_opendid
+title: Integrate OpenDID
+---
+
+OpenDID follows the [OpenID Connect 1.0 Specification](https://openid.net/specs/openid-connect-core-1_0.html#Introduction) and implements both the [implicit flow](https://openid.net/specs/openid-connect-core-1_0.html#ImplicitFlowSteps)
+and the [authorization code flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth).
+Read the [demo project guide](05_demo_project.md) for an example of integrating OpenDID.
+
+## Authorization code flow
+
+Initiate the flow by redirecting to the **GET** `/api/v1/authorize` endpoint on the OpenDID service and setting the following query
+parameters:
+
+-   `response_type`: set value to `code` to indicate Authorization Code Flow.
+-   `client_id`: The client ID set in the config.yaml file.
+-   `redirect_uri`: OpenDID will redirect to this URL after authentication.
+-   `scope`: set value to `openid`.
+-   `state`: set to a secure random number.
+-   `nonce`: optional value, set to a secure random number.
+
+**Example**:
+
+```
+GET /api/v1/authorize?
+    response_type=code&
+    client_id=example-client&
+    redirect_uri=http://localhost:1606/callback.html&
+    scope=openid&
+    state=rkw49cbvd4azu5dsln1xbl&
+    nonce=vedur4om49ei8w91jt7wt HTTP/1.1
+```
+
+After successful authentication, the OpenDID service redirects back to the provided `redirect_uri` with `code` and `state` query parameters.
+
+**Example**:
+
+```
+/callback.html?
+    code=lwDS1ZpQBwR4Vdm53_L8bWpUJ1mx9A0mA_-86dubTqzqzwGazx1RyLX4Z_qf&
+    state=rkw49cbvd4azu5dsln1xbl
+```
+
+You can retrieve the `id_token` by calling the **POST** `/api/v1/token` and providing the following values in the form serialization:
+
+-   `code`: code value returned from `authorize`.
+-   `grant_type`: set value to `authorization_code`.
+-   `redirect_uri`: the same `redirect_uri` used in `authorize`.
+-   `client_id`: the client ID set in the `config.yaml` file.
+-   `client_secret`: the client secret value set in the `config.yaml` file.
+
+**Example**:
+
+```
+POST /api/v1/token HTTP/1.1
+Content-Type: application/x-www-form-urlencoded
+
+code=lwDS1ZpQBwR4Vdm53_L8bWpUJ1mx9A0mA_-86dubTqzqzwGazx1RyLX4Z_qf&
+grant_type=authorization_code&
+redirect_uri=http%3A%2F%2Flocalhost%3A1606%2Fcallback.html&
+client_id=example-client&
+client_secret=insecure_client_secret
+```
+
```

Comment:
Maybe adding a shell command to execute the POST request. Makes it faster for the developer to get the full experience of the Authorization code flow.
```
curl -X POST http://localhost:3001/api/v1/token \
  -H "Content-Type: application/x-www-form-urlencoded" \
  -d "code=lwDS1ZpQBwR4Vdm53_L8bWpUJ1mx9A0mA_-86dubTqzqzwGazx1RyLX4Z_qf" \
  -d "grant_type=authorization_code" \
  -d "redirect_uri=http://localhost:1606/callback.html" \
  -d "client_id=example-client" \
  -d "client_secret=insecure_client_secret"
```
Note that the code value needs to be changed.

Reply:
We probably can leave it as is, since the token endpoint is just one part of multiple requests and interactions with the extension, and these are not possible to do with curl.